cpp/src/arrow/array/builder_primitive.h

```diff
@@ -146,6 +146,11 @@ class NumericBuilder : public ArrayBuilder {
     return reinterpret_cast<value_type*>(data_builder_.mutable_data())[index];
   }
 
+  void UnsafeSetIsNull(int64_t index, bool is_null) {
+    this->operator[](index) = value_type{};
```

Comment:
It may cause surprise if this api always clears an existing slot implicitly, even if we valid-fy a valid slot, which I suppose should have no side effect.

Reply:
Yeah. Here I am trying to imitate what `AppendNull` does.